flowcraft/generator/templates/quast.nf

```diff
@@ -0,0 +1,48 @@
+if (params.reference{{param_id}} == null && params.genomeSize{{param_id}} == null)
+    exit 1, "Specify at least one of reference or genomeSize"
+if (params.reference{{param_id}} != null && params.genomeSize{{param_id}} != null)
+    exit 1, "Specify only one of reference or genomeSize"
+
+if (params.reference{{param_id}} != null) {
+    process quast_{{pid}} {
+        {% include "post.txt" ignore missing %}
+
+        tag { sample_id }
+        publishDir "results/assembly/quast_{{pid}}/$sample_id", pattern: "*.tsv"
+        publishDir "reports/assembly/quast_{{pid}}/$sample_id"
+
+        input:
+        set sample_id, file(assembly) from {{input_channel}}
+        file reference from Channel.fromPath(params.reference{{param_id}})
+
+        output:
+        file "*"
+        {% with task_name="quast" %}
+        {%- include "compiler_channels.txt" ignore missing -%}
+        {% endwith %}
+
+        script:
+        "/usr/bin/time -v quast -o . -r $reference -s $assembly -l $sample_id -t $task.cpus >> .command.log 2>&1"
```

Comment:
There's no trouble in using `time` to check the run time of the software, but that information (and more) will already be available on the `pipeline_stats.txt` file generated during the execution of the pipeline :smile:

Reply:
That's excellent!

This is odd though. `/usr/bin/time -v` reports
```
Maximum resident set size (kbytes): 408240
```
whereas `pipeline_stats.txt` reports
```
rss = 58.9 MB
vmem = 476.4 MB
```
So I'd like to leave `/usr/bin/time -v` in until I sort out this discrepancy.

Reply:
No problem. Though that's interesting. It could be that the `rss` reported in by nextflow takes into account any possible overhead of a docker/singularity execution, while the `time` one is purely from quast. Or they simply measure `rss` differently. Either way I would be interesting to find out why!

Reply:
That may be. I doesn't oddly, which is why I usually use `zsh -c 'time foobar'` rather than `/usr/bin/time` for this purpose.

Reply:
I don't think `zsh` is in that `quast` Docker image though.